storage/storage.go

```diff
@@ -0,0 +1,99 @@
+package storage
+
+import (
+	"errors"
+	"go-http/models"
+	"sync"
+)
+
+type Adder interface {
+	Add(user *models.User) error
+}
+
+type Updater interface {
+	UpdateOne(username string, user *models.User) (*models.User, error)
+}
+
+type Getter interface {
+	GetAll() []*models.User
+}
+
+type GetterOne interface {
+	GetOne(username string) (*models.User, error)
+}
+
+type Deleter interface {
+	DeleteUser(username string) error
+}
+
+type Users struct {
+	mx      sync.RWMutex
+	storage map[string]*models.User
+}
+
+func New() *Users {
+	return &Users{
+		storage: make(map[string]*models.User),
+	}
+}
+
+func (users *Users) Add(user *models.User) error {
+	users.mx.Lock()
+	defer users.mx.Unlock()
+	if _, ok := users.storage[user.Username]; ok {
+		return errors.New("user already exists")
```

Comment:
Errors usually go to the top the file and exported as variables in order for someone to check for specific error. So your error should go in the `var ErrAlreadyExists = errors.New("user already exists")`. Then when i call your `Add` method i can check against that particular error like this:
```go

err := storage.Add(&user)

if err == storage.ErrAlreadyExists {
  // do my thing here
}

```